test/extended/util/framework.go

```diff
@@ -56,18 +61,104 @@ func WaitForInternalRegistryHostname(oc *CLI) (string, error) {
 		imageConfig, err := oc.AsAdmin().AdminConfigClient().ConfigV1().Images().Get("cluster", metav1.GetOptions{})
 		if err != nil {
 			if errors.IsNotFound(err) {
+				e2e.Logf("Image config object not found")
 				return false, nil
 			}
+			e2e.Logf("Error accessing image config object: %#v", err)
 			return false, err
 		}
 		if imageConfig == nil {
+			e2e.Logf("Image config object nil")
 			return false, nil
 		}
 		registryHostname = imageConfig.Status.InternalRegistryHostname
 		if len(registryHostname) == 0 {
+			e2e.Logf("Internal Registry Hostname is not set in image config object")
 			return false, nil
 		}
-		return true, nil
+
+		// verify that the OCM config's internal registry hostname matches
+		// the image config's internal registry hostname
+		ocm, err := oc.AdminOperatorClient().OperatorV1().OpenShiftControllerManagers().Get("cluster", metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		observedConfig := map[string]interface{}{}
+		err = json.Unmarshal(ocm.Spec.ObservedConfig.Raw, &observedConfig)
+		if err != nil {
+			return false, nil
+		}
+		internalRegistryHostnamePath := []string{"dockerPullSecret", "internalRegistryHostname"}
+		currentRegistryHostname, _, err := unstructured.NestedString(observedConfig, internalRegistryHostnamePath...)
+		if err != nil {
+			e2e.Logf("error procesing observed config %#v", err)
+			return false, nil
+		}
+		if currentRegistryHostname != registryHostname {
+			e2e.Logf("OCM observed config hostname %s does not match image config hostname %s", currentRegistryHostname, registryHostname)
+			return false, nil
+		}
+		// check pod logs for messages around image config's internal registry hostname has been observed and
+		// and that the build controller was started after that observation
+		pods, err := oc.AdminKubeClient().CoreV1().Pods("openshift-controller-manager").List(metav1.ListOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		foundOCMLogs := false
+		podLogs := map[string]string{}
+		for _, pod := range pods.Items {
+			req := oc.AdminKubeClient().CoreV1().Pods("openshift-controller-manager").GetLogs(pod.Name, &corev1.PodLogOptions{})
+			readCloser, err := req.Stream()
+			if err == nil {
+				b, err := ioutil.ReadAll(readCloser)
+				if err == nil {
+					podLog := string(b)
+					podLogs[pod.Name] = podLog
+					scanner := bufio.NewScanner(strings.NewReader(podLog))
+					firstLog := false
+					for scanner.Scan() {
+						line := scanner.Text()
+						if strings.Contains(line, "docker_registry_service.go") && strings.Contains(line, registryHostname) {
```

Comment:
does this output get printed in the OCM w/o increasing the OCM's verbosity level?  this is why i was thinking we might need to make it an info message (which, again, I don't think is a bad thing in general for helping us investigate customer clusters)

Reply:
that comes from the standard verbosity level

Reply:
sorry, i did a bad grep.

Reply:
no worries